snafu/uperf_wrapper/trigger_uperf.py

```diff
@@ -91,18 +87,21 @@ def _run_uperf(self):
 
     def _parse_stdout(self, stdout):
         # This will effectivly give us:
-        # ripsaw-test-stream-udp-16384
-        config = re.findall(r"running profile:(.*) \.\.\.", stdout)
-        test = re.split("-", config[0])[0]
-        protocol = re.split("-", config[0])[1]
-        size = re.split("-", config[0])[2]
-        nthr = re.split("-", config[0])[3]
+        # <profile name="{{test}}-{{proto}}-{{wsize}}-{{rsize}}-{{nthr}}">
+        config = re.findall(r"running profile:(.*) \.\.\.", stdout)[0]
+        test_type, protocol, wsize, rsize, nthr = config.split("-")
         # This will yeild us this structure :
         #     timestamp, number of bytes, number of operations
         # [('1559581000962.0330', '0', '0'), ('1559581001962.8459', '4697358336', '286704') ]
         results = re.findall(r"timestamp_ms:(.*) name:Txn2 nr_bytes:(.*) nr_ops:(.*)", stdout)
-        return {"test": test, "protocol": protocol, "message_size": size, "num_threads": nthr,
-                "results": results}
+        # We need this to don't break implementations depending on the previous non-asymmetric implementation
+        if wsize == rsize:
+            return results, {"test_type": test_type, "protocol": protocol, "message_size": int(rsize),
+                             "num_threads": int(nthr), "duration": len(results)}
+        else:
+            return results, {"test_type": test_type, "protocol": protocol, "write_message_size": int(wsize),
```

Comment:
Could we drop `write_` and have `message_size` and `read_message_size`? Can we make the position that `message_size` is write_message_size?